packages/models-library/src/models_library/projects.py

```diff
@@ -17,13 +18,24 @@
 from .projects_state import ProjectState
 from .projects_ui import StudyUI
 
-ProjectID = UUID
-ProjectIDStr = constr(regex=UUID_RE_BASE)
+ProjectID: TypeAlias = UUID
+ClassifierID: TypeAlias = str
 
-ClassifierID = str
+NodesDict: TypeAlias = dict[NodeIDStr, Node]
 
-# TODO: for some reason class Workbench(BaseModel): __root__= does not work as I thought ... investigate!
-Workbench = dict[NodeIDStr, Node]
+
+class ProjectIDStr(ConstrainedStr):
+    regex = re.compile(UUID_RE_BASE)
+
+    class Config:
+        frozen = True
+
+
+class DateTimeStr(ConstrainedStr):
+    regex = re.compile(DATE_RE)
```

Comment:
what do we use here? we should go with some ISO if possible.

Reply:
see response below